### PR TITLE
Develop locally without opening public tunnel

### PR DESCRIPTION
**Pull request by jimmoffet**
**TLDR: Spoof incoming messages from slack instead of opening a public tunnel**

This doesn't require any incoming communication from slack servers, but the bolt app object does do an outgoing handshake when it wakes up, so you need a valid signing secret and bot token stored locally. Could probably spoof this handshake too, and fully isolate the local app, if needed. Waiting for a sanity check on whether any part of this PR is necessary/useful.

In spoof-message.js, we compute a valid signature that bolt will accept and construct a request that simulates the requests sent by slack's servers, which we POST to ://localhost/slack/events via axios.

**Steps to develop locally without a public tunnel (can also be done inside docker with extra debugger config):**

Run (only) pg in a container
```
docker compose -f database.local.yml up --build
```

Run charlie node outside container
```
npm start dev
```

Spoof a message to charlie
```
node -e 'require("./src/spoof-message").spoofMessage("dog fact or some similar message")'
```

At this point, one should be able to use any of the usual local node debugging tools/techniques. The primary intention here is to be able to use debugging niceties (breakpoints, etc...) that might be difficult to use when pushing to a remote instance.

**Note:** still a draft, needs testing and I don't think I've tried to push a slash command through it at all yet!

### CODE REVIEW

File: src/spoof-message.js [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+
+function spoofMessage(
+  msg,
+  token = "52YK4N4phElR07An4dC59yMR", // optional
+  teamID = "TUMHGJ48G", // optional
+  channel = "CUM5CUN5A", // optional
+  apiAppID = "A02N5BNE13L", // optional
+  userID = "UUPDE22F9", // optional
+  eventID = "Ev03HC9TGA86", // optional
+  authUserID = "U02N2D5FRB7", // optional
```
Comment (mgwalker) [inline src/spoof-message.js:46]: Are these tokens just made up, as samples? Just confirming that they aren't real values that could lead to a compromise. 😬
